### PR TITLE
[mlir] update remaining transform tests to main pass

### PR DESCRIPTION
**Pull request by ftynse**
Use the main transform interpreter pass instead of the test pass. The only tests that are not updated are specific to the operation of the test pass.

### CODE REVIEW

File: mlir/test/Dialect/Transform/test-interpreter-debug.mlir [hunk 1]
```diff
@@ -1,62 +1,38 @@
...
-}
-
-// -----
-
-module {
-  // expected-error @below {{expected the transform entry point to be a top-level transform op}}
-  func.func private @foo() attributes {transform.target_tag="transform"}
-
-  module attributes {transform.target_tag="payload"} {}
-}
```
Comment (martin-luecke) [inline mlir/test/Dialect/Transform/test-interpreter-debug.mlir:46]: Do these tests not make sense anymore for the interpreter pass?

**Review by martin-luecke — APPROVED** [review]
LGTM

Just a few short questions:

Is there documentation somewhere on how the trailingArgs need to be structured and which types of args are supported?

What is the difference between:
`debug-bind-trailing-args=func.func`
and
`debug-bind-trailing-args=^func.func`?

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

File: mlir/test/Dialect/Transform/test-interpreter-debug.mlir (hunk 1 above)
Comment (ftynse) [reply mlir/test/Dialect/Transform/test-interpreter-debug.mlir:46]: No, the new pass is not using discardable attributes to identify the entry point.

**Comment by ftynse** [comment]
> Is there documentation somewhere on how the trailingArgs need to be structured and which types of args are supported?

Added more ample documentation to the pass description.

**Pull request merged**